README.md

```diff
@@ -1,19 +1,96 @@
 # Submit a GAR Public Site UI
 
-## Docker Usage
-The following depends on a .env file in the root of this project with the required environment variables. This will start a local test instance of keycloak for local testing.
+## Requirements
+- docker
+- docker-compose
+- node
+- npm
+- your email added to GOV Notify Dev whitelist by repository administrator
+
+## Development
+
+The following depends on a `.env.dev` file in the root of this project with
+the required environment variables. To obtain it please contact repository administrator.
+
+This will start local instances of:
+- Database (used for both Frontend and Backend)
+- API (you need a local copy of `api` docker image loaded, 
+       please contact repository administrators to obtain it)
+- ClamAV engine
+- ClamAV REST API 
+- Frontend
+
+Please note - due to security configuration, the following lines need to be 
+commented out for local development, but not committed back to the repository:
+```
+./src/server.js (lines 95-98):
+  // app.use(csrf({cookie: {
+  //   httpOnly: true,
+  //   secure: true
+  // }}));
+
+./src/server.js (lines 103-106):
+    // var token = req.csrfToken();
+    // res.cookie('XSRF-TOKEN', token);
+    //
+    // res.locals.csrfToken = token;
+```
+
+
+To start clean:
+```
+docker-compose up --build -d
+```
+These steps will create 5 endpoints:
+- http://localhost:5432 -> Database
+- http://localhost:5000 -> API
+- http://localhost:3310 -> ClamAV engine
+- http://localhost:8080 -> ClamAV REST API
+- http://localhost:3000 -> Node site
+
+
+Once it's up, you need to:
+- whitelist yourself in database:
+```
+for line in $(egrep -v "^#|^$" .env.dev); do export $line; done
+docker exec -ti database sh -c "PGPASSWORD=${POSTGRES_PASSWORD} psql -U ${POSTGRES_USER} ${POSTGRES_DB}"
+```
+Once inside database REPL run these commands. 
+Don't forget to update your email address in the INSERT command:
+```
+CREATE TABLE "session" (
+  "sid" varchar NOT NULL COLLATE "default",
+	"sess" json NOT NULL,
+	"expire" timestamp(6) NOT NULL
+)
+WITH (OIDS=FALSE);
+ALTER TABLE "session" ADD CONSTRAINT "session_pkey" PRIMARY KEY ("sid") NOT DEFERRABLE INITIALLY IMMEDIATE;
+
+INSERT INTO "WhiteList" ("email", "createdAt", "updatedAt") VALUES ('your@email.com', current_timestamp, current_timestamp);
+```
+
+
 
+To stop containers without cleaning data:
 ```
-docker-compose down -t 1
-docker-compose build
-docker-compose up -d
+docker-compose stop
 ```
 
-These steps will create 4 endpoints:
-- https://127.0.0.1:10443 -> main site secured with unverified TLS certificate (ignore CERT INVALID errors)
-- http://127.0.0.1:3000 -> node site
-- http://127.0.0.1:3310 -> ClamAV endpoint
+To start stopped containers:
+```
+docker-compose start
+```
 
+To apply your changes to Frontend only:
+```
+docker-compose stop node
+docker-compose up --build -d node
+```
```

Comment:
Looks good Pawel, the only comment I have now is we should also have a command in README for running the tests locally. For a developer trying to run the local environment for a story, they will definitely try to run the tests so the command in README will help.

Reply:
Re Non Docker Usage

Please excuse my NOOB understanding of the system but if you don't use the Docker method wouldn't you need a local database etc available?

Reply:
That's correct @nigelhughes552. 
Non docker/docker-compose setup would require you to install database, compile API code and start it, install Clamav service, install nginx proxy and configure it to work with clamav and have node/npm.
Quite a lot of work comparing to 1 `docker-compose up --build - d` command 😉